### DISCUSSION TITLE
Implementing TensorFlowjs model in Ionic framework with Capacitor

### DISCUSSION THREAD

**Discussion by Tanush99** [vercel/next.js#18650] [Help]
Here is my typescript code:

import { Component, OnInit, ViewChild, ElementRef, Renderer2 } from '@angular/core';
import { Plugins, CameraResultType, CameraSource} from '@capacitor/core';
import { DomSanitizer, SafeResourceUrl} from '@angular/platform-browser';
import { Platform } from '@ionic/angular';
import * as tf from '@tensorflow/tfjs';
import * as tr from '@tensorflow/tfjs-node';
import { rendererTypeName } from '@angular/compiler';
import { Base64 } from '@ionic-native/base64/ngx';
import { defineCustomElements } from '@ionic/pwa-elements/loader';
import { Buffer } from 'buffer';
 
 
const { Camera } = Plugins;
 
@Component({
  selector: 'app-predict',
  templateUrl: './predict.page.html',
  styleUrls: ['./predict.page.scss'],
})
export class PredictPage{
 
  linearModel : tf.Sequential;
  prediction : any;
  InputTaken : any;
  ctx: CanvasRenderingContext2D;
  pos = { x: 0, y: 0 };
  canvasElement : any;
  photo: SafeResourceUrl;
  model: tf.LayersModel;
 
  constructor(public el : ElementRef , public renderer : Renderer2 , public platform : Platform, private base64: Base64,
    private sanitizer: DomSanitizer) 
  {}
 
  
  async takePicture() {
  
    const image = await Camera.getPhoto({
        quality: 90,
        allowEditing: true,
        resultType: CameraResultType.DataUrl,
        source: CameraSource.Camera});
      const model = await tf.loadLayersModel('src/app/assets/model.json');
      this.photo = this.sanitizer.bypassSecurityTrustResourceUrl(image.base64String);
      const b = Buffer.from(image.base64String, 'base64')
      defineCustomElements(window);
    const pred = await tf.tidy(() => {
      // Move the base64 image into a buffer
 
      // get the tensor
    const image = tr.node.decodeImage(b)
 
      // Compute the output
    const output = this.model.predict(image) as any;
 
    // Save predictions on the component
    this.prediction = Array.from(output.dataSync());
      
    
        
    });
  }
}

Here, in the above code, I have imported the necessary tools. Then, I have my constructor function and a takepicture() function. In the takepicture function, I have included functionality for the user to take pictures. I have converted the pictures to tensor form to pass to model.predict. Then, I will use html to display the prediction. However, when I import buffer, I get the following error: 

[C:\Users\isusu\Documents\first\fs]
[C:\Users\isusu\Documents\first\fs.ts]
[C:\Users\isusu\Documents\first\fs.tsx]
[C:\Users\isusu\Documents\first\fs.mjs]
[C:\Users\isusu\Documents\first\fs.js]
[C:\Users\isusu\Documents\first\node_modules\ignore-walk\node_modules]
[C:\Users\isusu\Documents\first\node_modules\node_modules]
[C:\Users\isusu\Documents\node_modules]
[C:\Users\isusu\node_modules]
[C:\Users\node_modules]
[C:\node_modules]
[C:\Users\isusu\Documents\first\node_modules\fs]
[C:\Users\isusu\Documents\first\node_modules\fs.ts]
[C:\Users\isusu\Documents\first\node_modules\fs.tsx]
[C:\Users\isusu\Documents\first\node_modules\fs.mjs]
[C:\Users\isusu\Documents\first\node_modules\fs.js]
 @ ./node_modules/ignore-walk/index.js 3:11-24
 @ ./node_modules/npm-packlist/index.js
 @ ./node_modules/node-pre-gyp/lib/package.js
 @ ./node_modules/node-pre-gyp/lib sync ^\.\/.*$
 @ ./node_modules/node-pre-gyp/lib/node-pre-gyp.js
 @ ./node_modules/@tensorflow/tfjs-node/dist/index.js
 @ ./src/app/predict/predict.page.ts
 @ ./src/app/predict/predict.module.ts
 @ ./src/app/app-routing.module.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/mkdirp/index.js
Module not found: Error: Can't resolve 'fs' in 'C:\Users\isusu\Documents\first\node_modules\mkdirp'
resolve 'fs' in 'C:\Users\isusu\Documents\first\node_modules\mkdirp'
  Parsed request is a module
  using description file: C:\Users\isusu\Documents\first\node_modules\mkdirp\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in C:/Users/isusu/Documents/first
        using description file: C:\Users\isusu\Documents\first\package.json (relative path: .)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: C:\Users\isusu\Documents\first\package.json (relative path: ./fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\fs.tsx doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\fs.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\fs.js doesn't exist
            as directory
              C:\Users\isusu\Documents\first\fs doesn't exist
      C:\Users\isusu\Documents\first\node_modules\mkdirp\node_modules doesn't exist or is not a directory
      C:\Users\isusu\Documents\first\node_modules\node_modules doesn't exist or is not a directory
      C:\Users\isusu\Documents\node_modules doesn't exist or is not a directory
      C:\Users\isusu\node_modules doesn't exist or is not a directory
      C:\Users\node_modules doesn't exist or is not a directory
      C:\node_modules doesn't exist or is not a directory
      looking for modules in C:\Users\isusu\Documents\first\node_modules
        using description file: C:\Users\isusu\Documents\first\package.json (relative path: ./node_modules)
          Field 'browser' doesn't contain a valid alias configuration
          using description file: C:\Users\isusu\Documents\first\package.json (relative path: ./node_modules/fs)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\node_modules\fs doesn't exist
            .ts
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\node_modules\fs.ts doesn't exist
            .tsx
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\node_modules\fs.tsx doesn't exist
            .mjs
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\node_modules\fs.mjs doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\isusu\Documents\first\node_modules\fs.js doesn't exist
            as directory
              C:\Users\isusu\Documents\first\node_modules\fs doesn't exist
It would be appreciated if I could receive some guidance regarding this topic as there are not many resources online that could help me with this issue.

**Comment by gummyWalrus** [comment]
Please use github markdown format. 
For example i can format typescript code this way.
```ts
console.log("Nicely formatted !");
```

To do this use 3 backticks \`\`\` then the name of the language input a newline, write your code, then 3 backticks again

for example
```markdown
```typescript
 console.log('Now people will read my code with ease :) !');
 \```
```
Dont write the backslash at the end.

Of course i'll try to help you the best I can once I'll be able to read your code.
